cf_xarray/accessor.py

```diff
@@ -1312,18 +1310,21 @@ def rename_like(
         ourkeys = self.keys()
         theirkeys = other.cf.keys()
 
-        good_keys = set(_COORD_NAMES) & ourkeys & theirkeys
-        if not good_keys:
-            raise ValueError(
-                "No common coordinate variables between these two objects."
-            )
-
+        good_keys = ourkeys & theirkeys
         renamer = {}
         for key in good_keys:
-            ours = _get_axis_coord_single(self._obj, key)[0]
-            theirs = _get_axis_coord_single(other, key)[0]
-            renamer[ours] = theirs
+            ours = apply_mapper(
+                (_get_axis_coord, _get_with_standard_name), self._obj, key, error=False
+            )
+            theirs = apply_mapper(
+                (_get_axis_coord, _get_with_standard_name), other, key, error=False
+            )
+            if len(ours) > 1 or len(theirs) > 1:
```

Comment:
Could optionally raise a warning here if `verbose=True`

Reply:
Maybe warn ant the end about all `other.variables` missing in `renamer.values()`?

Reply:
This might work out with staggered grids: Rename axes first, if `len(ours)>1 or len(theirs)>1` and all variables in `ours` and `theirs` have different combinations of dimensions we should be able to rename. As we already renamed the dimensions, we just have to look at their dims.